orion/pkg/controllers/location_controller.go

```diff
@@ -63,8 +64,14 @@ func PublishLocations(c *gin.Context) {
 
 	errorList := services.LocationService.Publish(locIdsJson)
 	if len(errorList) > 0 {
-		c.Error(errors.New("one or more locations failed to publish"))
-		c.JSON(http.StatusInternalServerError, errorList)
+		var errorStrings []string
+		errorStrings = append(errorStrings, "one or more locations failed to publish")
+		for _, errorBody := range errorList {
```

Comment:
Make a public function in `domains` package (you can put it in same file as PublicErrorBody) that concatenates all the errors here.
No need to replace this same code over and over.

So this whole part could look like:
```
err := concatErrors(errorList, "one or more ....")
c.Error(err)
c.String(...)
```